src/pages/index.js

```diff
@@ -0,0 +1,285 @@
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+// Delete form submission handler
+
+function handleDeleteSubmit(evt) {
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Deleting", "Delete");
+  evt.preventDefault();
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Deleting", "Delete");
+    });
+}
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+// Like button functionality
+
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-btn_liked");
+    })
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardDeleteBtn.addEventListener("click", (evt) =>
+    handleCardDelete(cardElement, data._id)
+  );
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+  return cardElement;
+}
+
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+function handleEditProfileSubmit(evt) {
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  evt.preventDefault();
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      disableButton(
+        editModal.querySelector(settings.submitButtonSelector),
+        settings
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.